crates/core/src/metadata/manager.rs

```diff
@@ -248,6 +275,27 @@ impl MetadataManager {
             inbound: metadata_builder.receiver,
             metadata_store_client,
             update_tasks: EnumMap::default(),
+            metadata_mask: EnumSet::all(),
+        }
+    }
+
+    pub fn new_with_roles(
```

Comment:
@tillrohrmann what do you think about a constructor like this that will allow us to optimize the MetadataManager runtime behavior by role set? We still potentially sync too much data more frequently than necessary, but at least it will not involve nodes that don't care about it. If we like this approach, I can wire this up in the node initialization path.